src/osemgrep/cli_scan/Scan_CLI.ml

```diff
@@ -876,25 +872,36 @@ let cmdline_term ~allow_empty_config : conf Term.t =
       | _else_ -> default.output_format
     in
     let engine_type =
-      match (oss, pro_lang, pro_intrafile, pro) with
-      | false, false, false, false when secrets ->
-          Engine_type.(PRO Language_only)
-      | false, false, false, false -> default.engine_type
-      | true, false, false, false when secrets ->
-          Error.abort
-            "Mutually exclusive options --oss/--beta-testing-secrets-enabled"
-      | true, false, false, false -> OSS
-      | false, true, false, false -> PRO Engine_type.Language_only
-      | false, false, true, false -> PRO Engine_type.Intrafile
-      | false, false, false, true -> PRO Engine_type.Interfile
-      | _else_ ->
-          (* TOPORT: list the possibilities *)
-          Error.abort
-            "Mutually exclusive options \
-             --oss/--pro-languages/--pro-intrafile/--pro"
+      (* This first bit just rules out mutually exclusive options. *)
+      if oss && secrets then
+        Error.abort
+          "Mutually exclusive options --oss/--beta-testing-secrets-enabled";
+      if
+        [ oss; pro_lang; pro_intrafile; pro ]
+        |> Common2.filter Fun.id |> List.length > 1
+      then
+        Error.abort
+          "Mutually exclusive options \
+           --oss/--pro-languages/--pro-intrafile/--pro";
+      (* Now select the engine type *)
+      if oss then Engine_type.OSS
+      else
+        let analysis =
+          Engine_type.(
+            match () with
+            | _ when pro_intrafile -> Deep_intrafile
+            | _ when pro -> Deep_interfile
+            | _ when pro_lang -> OSS_intrafile
+            | _ -> OSS_intrafile)
+        in
+        let extra_languages = pro || pro_lang || pro_intrafile in
+        let secrets_config =
+          if secrets && not no_secrets_validation then
+            Some Engine_type.{ allow_all_origins = allow_untrusted_validators }
+          else None
+        in
+        Engine_type.make ~extra_languages ~analysis ~secrets_config
```

Comment:
maybe add a small comment saying (* note that this can still lead to Engine_type.OSS when no option was set *).
Personally I would remove the Engine_type.make 
and instead build directly Pro or OSS here. Adding an extra layer of function complicates even more the logic.